aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -99,181 +106,167 @@ The preceding command adds the [aspnet-codegenerator scaffolding tool](xref:fund
-* Select **Add**
+<!-- End of tabs                  -->
 
-Visual Studio creates:
+Scaffolding creates:
 
-* A movies controller (*Controllers/MoviesController.cs*)
-* Razor view files for Create, Delete, Details, Edit, and Index pages (*Views/Movies/\*.cshtml*)
+* A movies controller: *Controllers/MoviesController.cs*.
+* Razor view files for Create, Delete, Details, Edit, and Index pages: `Views/Movies/\*.cshtml`.
+* A database context class: *Data/MvcMovieContext.cs*.
 
 The automatic creation of these files is known as *scaffolding*.
 
-### [Visual Studio Code](#tab/visual-studio-code) 
+The scaffolded pages can't be used yet because the database doesn't exist. Running the app and selecting the **Movie App** link results in a *Cannot open database* or *no such table: Movie* error message.
 
-* Open a command window in the project directory (The directory that contains the *Program.cs*, *Startup.cs*, and *.csproj* files).
+<a name="dc"></a>
 
-* On Linux, export the scaffold tool path:
+### Examine the generated database context class
 
-  ```console
-  export PATH=$HOME/.dotnet/tools:$PATH
-  ```
+A database context class is needed to coordinate EF Core functionality (Create, Read, Update, Delete) for the `Movie` model. The database context is derived from [Microsoft.EntityFrameworkCore.DbContext](xref:Microsoft.EntityFrameworkCore.DbContext) and specifies the entities to include in the data model.
 
-* Run the following command:
+Scaffolding created the following database context class: Data/MvcMovieContext.cs:
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/zDocOnly/MvcMovieContext.cs?name=snippet)]
 
-  ```dotnetcli
-  dotnet aspnet-codegenerator controller -name MoviesController -m Movie -dc MvcMovieContext --relativeFolderPath Controllers --useDefaultLayout --referenceScriptLibraries
-  ```
+The preceding code creates a [DbSet\<Movie>](/dotnet/api/microsoft.entityframeworkcore.dbset-1) property for the entity set. In Entity Framework terminology, an entity set typically corresponds to a database table. An entity corresponds to a row in the table.
 
-  [!INCLUDE [explains scaffold generated params](~/includes/mvc-intro/model4.md)]
+<a name="reg"></a>
 
-### [Visual Studio for Mac](#tab/visual-studio-mac)
+### Examine the generated database context registration
 
-* Open a command window in the project directory (The directory that contains the *Program.cs*, *Startup.cs*, and *.csproj* files).
+ASP.NET Core is built with [dependency injection (DI)](xref:fundamentals/dependency-injection). Services, such as the EF Core DB context, must be registered with DI in `Startup`. Components that require these services are provided these services via constructor parameters. For example, Razor Pages frequently requires services from the DI container. The constructor code that gets a DB context instance is shown later in the tutorial. In this section, the database context is registered with the DI container.
```

Comment:
Why are you talking about RP?

Reply:
In the following code, the movie cntx is ...

Reply:
copy paste error, I forgot to remove the the mention of RP as the example.  Thanks for catching this.

Reply:
Could you be more explicit about what you to do with this: "In the following code, the movie cntx is ."  I'm sorry I am not catching on.  Sometimes brevity is not a friend to me here.     I can assume "cntx" means "context", but could you clarify what what you are suggesting here?

Reply:
Replace

> The constructor code that gets a DB context instance is shown later in the tutorial. In this section, the database context is registered with the DI container.

with text code found later in the doc

In the Controllers/MoviesController.cs file, The constructor uses Dependency Injection to inject the MvcMovieContext database context into the controller. The database context is used in each of the CRUD methods in the controller.

Then remove the following section

# Dependency injection in the controller